File: Apps/DriveSwitch/app/src/main/java/com/driveswitch/Event.java
package com.driveswitch;


import java.util.Date;

public class Event
{
    public Event(Double latitude, Double longitude)
    {
        _t = "Event";

        Name = "";
        TypeId = 0;
        DateTime = new Date();

        Reference = new Reference(Constants.EmptyObjectId, 0);
        Details = "None";

        Location = new Location(Constants.EmptyObjectId, Constants.EmptyObjectId, "", latitude, longitude);
    }
    public String _t;
    public String Name;
    public Integer TypeId;
    public Date DateTime ;
    public Reference Reference;
    public String Details;
    public Location Location;
}

File: WebApi/WebApi/Models/Event.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

using MongoDB;
using MongoDB.Bson;
using MongoDB.Driver;

namespace WebApi.Models
{
    public class Event : Base
    {
        public Event(Double latitude, Double longitude)
        {
            _t = "Event";

            Name = "";
            TypeId = 0;
            DateTime = DateTime.UtcNow;

            Reference = new Reference(ObjectId.Empty, 0);
            Details = "None";

            Location = new Location(ObjectId.Empty, ObjectId.Empty, "", latitude, longitude);
        }
        public int TypeId { get; set; }
        public DateTime DateTime { get; set; }
        public Reference Reference { get; set; }
        public string Details { get; set; }
        public Location Location { get; set; }
    }
}